air/src/options.rs

```diff
@@ -1,133 +1,48 @@
 
-    /// Creates a new instance of [ProvingOptions] from the specified parameters.
-    pub fn new(
-        num_queries: usize,
-        blowup_factor: usize,
-        grinding_factor: u32,
-        field_extension: FieldExtension,
-        fri_folding_factor: usize,
-        fri_remainder_max_degree: usize,
-        hash_fn: HashFunction,
-    ) -> Self {
-        let proof_options = WinterProofOptions::new(
-            num_queries,
-            blowup_factor,
-            grinding_factor,
-            field_extension,
-            fri_folding_factor,
-            fri_remainder_max_degree,
-            BatchingMethod::Algebraic,
-            BatchingMethod::Horner,
-        );
-        let exec_options = ExecutionOptions::default();
-        Self { exec_options, proof_options, hash_fn }
-    }
-
-    /// Creates a new preset instance of [ProvingOptions] targeting 96-bit security level, given
-    /// a choice of a hash function.
+    /// Creates a new instance of [ProvingOptions] with the specified hash function.
     ///
-    /// If the hash function is arithmetization-friendly then proofs will be generated using
-    /// settings that are well-suited for recursive verification.
-    pub fn with_96_bit_security(hash_fn: HashFunction) -> Self {
-        let proof_options = match hash_fn {
-            HashFunction::Blake3_192 | HashFunction::Blake3_256 => Self::REGULAR_96_BITS,
-            HashFunction::Rpo256 | HashFunction::Rpx256 | HashFunction::Poseidon2 => {
-                Self::RECURSIVE_96_BITS
-            },
-        };
+    /// The STARK proving parameters (security level, FRI config, etc.) are determined
+    /// by the hash function and hardcoded in the prover's config module.
+    pub fn new(hash_fn: HashFunction) -> Self {
         Self {
             exec_options: ExecutionOptions::default(),
-            proof_options,
             hash_fn,
         }
     }
 
-    /// Creates a new preset instance of [ProvingOptions] targeting 128-bit security level, given
-    /// a choice of a hash function, in the non-recursive setting.
+    /// Creates a new instance of [ProvingOptions] targeting 96-bit security level.
+    ///
+    /// Note: The actual security parameters are hardcoded in the prover's config module.
+    /// This is a convenience constructor that is equivalent to `new(hash_fn)`.
+    pub fn with_96_bit_security(hash_fn: HashFunction) -> Self {
+        Self::new(hash_fn)
+    }
+
+    /// Creates a new instance of [ProvingOptions] targeting 128-bit security level.
     ///
-    /// If the hash function is arithmetization-friendly then proofs will be generated using
-    /// settings that are well-suited for recursive verification.
+    /// Note: The actual security parameters are hardcoded in the prover's config module.
+    /// This is a convenience constructor that is equivalent to `new(hash_fn)`.
     pub fn with_128_bit_security(hash_fn: HashFunction) -> Self {
```

Comment:
AFAICT, we don't have a way to generate proofs at 128-bit security level any more - so, we should either remove this or add such options (could be done in a follow-up).